### PR TITLE
More stubs and header additions from #768 for systemd

### PR DESCRIPTION
**Pull request by Dennisbonke**
Batch number 2, again split off from #768. Nothing too big, a few function stubs and mostly kernel headers for the remaining part. Also adds one additional errno constant in the mlibc abi.

Part of the systemd on Managarm project.

### CODE REVIEW

File: options/posix/generic/posix_signal.cpp [hunk 1]
```diff
@@ -131,3 +131,8 @@ int sigaltstack(const stack_t *__restrict ss, stack_t *__restrict oss) {
 
 	return 0;
 }
+
+int sigisemptyset(const sigset_t *) {
+	__ensure(!"sigpending() not implemented");
```
Comment (qookei) [inline options/posix/generic/posix_signal.cpp:136]: Wrong message, also this is easy to implement (`return !(*set);` or so)

File: options/posix/include/bits/posix/posix_signal.h [hunk 2]
```diff
@@ -91,6 +91,8 @@ int sigtimedwait(const sigset_t *__restrict set, siginfo_t *__restrict info, con
 int sigwait(const sigset_t *__restrict set, int *__restrict sig);
 int sigwaitinfo(const sigset_t *__restrict set, siginfo_t *__restrict info);
 
+int sigisemptyset(const sigset_t *set);
```
Comment (qookei) [inline options/posix/include/bits/posix/posix_signal.h:94]: This is a GNU extension, maybe wrap in a check for the glibc option?

File: options/posix/generic/posix_signal.cpp (hunk 1 above)
Comment (Dennisbonke) [reply options/posix/generic/posix_signal.cpp]: Should be fixed.

File: options/posix/include/bits/posix/posix_signal.h (hunk 2 above)
Comment (Dennisbonke) [reply options/posix/include/bits/posix/posix_signal.h:97]: Good catch, wrapped as requested.

File: options/linux-headers/include/mtd/mtd-user.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	uint8_t type;
+	uint32_t flags;
+	uint32_t size;
+	uint32_t erasesize;
+	uint32_t writesize;
+	uint32_t oobsize;
+	uint64_t padding;
+};
+
+#define MEMGETINFO _IOR('M', 1, struct mtd_info_user)
```
Comment (64) [inline options/linux-headers/include/mtd/mtd-user.h:19]: does this work

**Review by 64 — APPROVED** [review]

File: options/linux-headers/include/mtd/mtd-user.h (hunk 3 above)
Comment (Dennisbonke) [reply options/linux-headers/include/mtd/mtd-user.h:19]: As in, is this define okay? Should be, matches Linux.

**Pull request merged**